### DISCUSSION TITLE
Add a Framework entry for AuthModules

### DISCUSSION THREAD

**Discussion by niccord** [RotherOSS/otobo#1548] [Ideas]
Hi everyone,
since a very early version of OTRS, `AuthModule` can be configured via System Configuration.
The same cannot be said for `AuthModule1`, `AuthModule2` and so on, and they have to be configured via `Kernel/Config.pm`.

Why is it so?
Can the support for AuthModule1 to 9 be added to Framework.xml?

The same written above is valid for `CustomerUser::AuthModule`.

Thanks.
Nicola

**Comment by svenoe** [comment]
Hi Nicola,

actually this is not quite right. Only `Customer::AuthModule` etc. can be seen, for the agents none is visible. One reason to keep things in Kernel/Config.pm is to enable a strict authorization separation between the system admin and the OTOBO/OTRS admin, which is why e.g. `DatabasePw` is not included, either. I would have to think a bit about different scenarios where this could be an issue with `AuthModule` (and really maybe including `Customer::AuthModule` is not such a good idea, either, keep in mind that even if information can not be changed when set in Config.pm it still can be seen in the frontend). Another reason though is, that with changing `AuthModule` as an OTOBO admin without system admin access, one error will shut you out of OTOBO and you won't have any way to correct that.

We quite certainly won't change the agent side of this. About the customer side I will have a discussion, if I have time, but I could imagine moving all options out of the SysConfig, too, here.

Sven

**Comment by svenoe** [comment]
I had a short discussion, and we would rather remove the `CustomerUser::AuthModule` from the SysConfig currently and make setting up both the customer as well as the agent authorization part of the installation process, to make that more accessible.

For people who really want to configure it via the SysConfig, adding a simple xml file at least is not that much work, although of course this doesn't make things "easier"...

I will convert this issue to a discussion. Currently this is not a thing we are planning on doing.
